bench/src/main/scala/cats/parse/bench/StringInBench.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.parse.bench
+
+import cats.parse.Parser
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+class StringInBenchmarks {
+  val text: String =
+    "foofoo"
+
+  val stringIn = Parser.stringIn1(List("foo", "bar", "foobar", "foofoo"))
+
+  val oneOf =
+    Parser.oneOf1(
+      List(
+        Parser.string1("foobar"),
+        Parser.string1("foofoo"),
+        Parser.string1("foo"),
+        Parser.string1("bar")
+      )
+    )
+
+  @Benchmark
+  def stringInParse(): Unit =
+    stringIn.parseAll(text) match {
+      case Right(()) => ()
+      case Left(e) => sys.error(e.toString)
```

Comment:
I think we should benchmark failures as well, since real parsers do have failures.

Reply:
Will do that